objects.go

```diff
@@ -18,9 +48,22 @@ type Blob struct {
 	obj core.Object
 }
 
-var ErrUnsupportedObject = errors.New("unsupported object type")
+// ID returns the object ID of the blob. The returned value will always match
+// the current value of Blob.Hash.
+//
+// ID is present to fufill the Object interface.
+func (b *Blob) ID() core.Hash {
```

Comment:
I really wanted to call this function `Hash()` so that the name matched `core.Object`'s equivalent, but that would have conflicted with the existing `Hash` field on `Commit`, `Tree`, `Blob` and `Tag`.